posthog-core/src/index.ts

```diff
@@ -1447,19 +1451,19 @@ export abstract class PostHogCore extends PostHogCoreStateless {
    *** GROUPS
    ***/
 
-  groups(groups: { [type: string]: string | number }): void {
+  groups(groups: PostHogGroupProperties): void {
     this.wrap(() => {
       // Get persisted groups
       const existingGroups = this.props.$groups || {}
 
       this.register({
         $groups: {
-          ...existingGroups,
+          ...(existingGroups as PostHogGroupProperties),
           ...groups,
         },
       })
 
-      if (Object.keys(groups).find((type) => existingGroups[type] !== groups[type])) {
+      if (Object.keys(groups).find((type) => existingGroups[type as keyof typeof existingGroups] !== groups[type])) {
```

Comment:
AI fix, have to test

Reply:
It's types, so if it passes CI, we're good